ethcore/machine/src/machine.rs

```diff
@@ -419,25 +419,6 @@ mod tests {
 		}
 	}
 
-	#[test]
-	fn should_disallow_unsigned_transactions() {
```

Comment:
I think we should keep the test. I guess the failure is going to be at a dfiferent level now, but still.

Reply:
we have this https://github.com/paritytech/parity-ethereum/pull/11335/files#diff-ffdd27c0e6928e872d31ecdc8f8e09eaR643

Reply:
or did you mean something else?

Reply:
Well, I just thing it worth to keep the test in it's current form, as it's covering a bit more than a constructor of `SignedTransaction`. I think it was added based on an actual previous bug, so keeping it as a way to prevent regressions should hurt.

Reply:
This test was added in https://github.com/paritytech/parity-ethereum/pull/8802 and it doesn't make sense anymore. I removed it because it actually fails on this branch, since it creates a null-signed `UnverifiedTransaction` and calls `verify_transaction_basic` expecting it to reject it.
But this PR removes this rejection logic, since it's not needed as we don't allow `SignedTransaction`s with null-signatures.
Now if I modify the test to convert a transaction to a `SignedTransaction`, then it's not different than the test linked above.

Reply:
The conversion happens within `engine/machine`, doesn't it (via `verify_transaction`). IMHO still worth to check the machine/engine integration with this, but not super strong on this. If you disagree - then fine, let's remove it.

Reply:
The conversion happens in the verification queue when we convert `Unverified` block to `PreverifiedBlock`
https://github.com/paritytech/parity-ethereum/blob/856a0755888a30d4dc52a68cb2638a8bfd5786ac/ethcore/verification/src/verification.rs#L100

Added dd4bbf3, let me know what you think.